### ISSUE TITLE
Migrate serde to use new stable implemntation

### ISSUE THREAD

**Issue by vitiral** [vitiral/artifact#29]
rust 1.15 allows for stable usage of serde. We should switch to that implementation ASAP

**Comment by vitiral** [comment]
we should wait on this until the toml library stabilizes

Blocked by alexcrichton/toml-rs#119

**Comment by vitiral** [comment]
this is done except the toml side. Will create a new issue for that later
